packages/multichain/src/scope/filter.ts

```diff
@@ -0,0 +1,115 @@
+import { type Hex } from '@metamask/utils';
+
+import { assertIsInternalScopeString, assertScopeSupported } from './assert';
+import { isSupportedMethod, isSupportedNotification } from './supported';
+import type {
+  InternalScopeString,
+  NormalizedScopeObject,
+  NormalizedScopesObject,
+} from './types';
+
+/**
+ * Groups a NormalizedScopesObject into two separate
+ * NormalizedScopesObject with supported scopes in one
+ * and unsupported scopes in the other.
+ * @param scopes - The NormalizedScopesObject to group.
+ * @param hooks - The hooks.
+ * @param hooks.isChainIdSupported - A helper that returns true if an eth chainId is currently supported by the wallet.
+ * @returns an object with two NormalizedScopesObjects separated by support.
+ */
+export const bucketScopesBySupport = (
+  scopes: NormalizedScopesObject,
+  {
+    isChainIdSupported,
+  }: {
+    isChainIdSupported: (chainId: Hex) => boolean;
+  },
+) => {
+  const supportedScopes: NormalizedScopesObject = {};
+  const unsupportedScopes: NormalizedScopesObject = {};
+
+  for (const [scopeString, scopeObject] of Object.entries(scopes)) {
+    assertIsInternalScopeString(scopeString);
+    try {
+      assertScopeSupported(scopeString, scopeObject, {
+        isChainIdSupported,
+      });
+      supportedScopes[scopeString] = scopeObject;
+    } catch (err) {
+      unsupportedScopes[scopeString] = scopeObject;
+    }
+  }
+
+  return { supportedScopes, unsupportedScopes };
+};
+
+/**
+ * Returns a NormalizedScopesObject with only
+ * scopes that are supported entirely as is.
+ * @param scopes - The NormalizedScopesObject to filter.
+ * @param hooks - The hooks.
+ * @param hooks.isChainIdSupported - A helper that returns true if an eth chainId is currently supported by the wallet.
+ * @returns a NormalizedScopesObject with only scopes that are currently supported.
+ */
+export const getSupportedScopes = (
```

Comment:
I can't find any uses of this 🤔

Reply:
This function was going to be used for helping to organize different categories of scopes in the `wallet_createSession` connection flow for when we enable adding scopes as part of the flow. Though looking at it now I'm not sure if its been superceded by `getSupportedScopeObject`... cc @jiexi ?

Reply:
Indeed it seems this is no longer used. I'll remove it